src/provider_completions.jl

```diff
@@ -1,7 +1,23 @@
 function process(r::JSONRPC.Request{Val{Symbol("textDocument/completion")},TextDocumentPositionParams}, server)
     tdpp = r.params
     line = get_line(tdpp, server)
-    comp = Base.REPLCompletions.completions(line, chr2ind(line,tdpp.position.character))[1]
+    io = IOBuffer(reverse(line[1:chr2ind(line,tdpp.position.character)]))
+    word = Char[]
```

Comment:
Actually, one should use an ``IOBuffer`` when constructing a ``String`` in many steps. So I would use an ``IOBuffer`` here, then convert that to a ``String`` using ``takebuf_string`` and then reverse the ``String``.

That should be much more efficient than using ``unshift!``.